nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -520,26 +521,22 @@ func updateUpgradeUnitGroups(upgradeClientSet *upgradeClientSet, d *schema.Resou
 		}
 	}
 
-	groupList, err := upgradeClientSet.GroupClient.List(nil, nil, nil, nil, nil, nil, nil, nil)
+	groupList, err := upgradeClientSet.GroupClient.List(&component, nil, nil, nil, nil, nil, nil, nil)
 	if err != nil {
 		return err
 	}
-	groupIDMap := make(map[string]bool)
-	groupNameMap := make(map[string]bool)
 
 	preUpgradeGroupID := ""
 	for _, groupI := range d.Get(componentToGroupKey[component]).([]interface{}) {
 		group := groupI.(map[string]interface{})
 		groupID := group["id"].(string)
 		var groupGet *model.UpgradeUnitGroup
-		var err error
 		isCreate := false
 		if groupID == "" {
```

Comment:
I would suggest to add a comment explaining what is the case for groups with ids and groups without